chaos-mesh/chaos-mesh
# Optional hostnetwork chaos setting

### What problem does this PR solve?

This brings back the ability to optionally run chaos testing on hostNetwork pods. I realize this is a dangerous option, but our tests depend on this functionality (We migrated our chaos tests to v1.0.0 that allowed this, but v1.0.1 onwards no longer allows it, breaking our tests) We intend on only using this for our temporary testing clusters and not to run this in production, ever.

### What is changed and how does it work?
This adds a switch to the chaos controller manager. 

```yaml
      containers:
      - command:
        - /usr/local/bin/chaos-controller-manager
        - -allowhostnetworktesting
```

When this is enabled it will allow testing of networkchaos on weave-pods or whatever has `hostNetwork: true`

### Checklist

Tests
No unit tests have been created yet, but I will add them if you are happy with me re-adding this behaviour.


Manual test:
- Install a new kind cluster as per the CONTRIBUTING.md 
- install the chaos-mesh infrastructure `./hack/local-up-chaos-mesh.sh`
- install a NetworkChaos test as follows:
```yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: NetworkChaos
metadata:
  name: network-bandwidth
  namespace: "kube-system"
spec:
  action: bandwidth
  mode: all
  namespaces:
  - "kube-system"
  selector:
    labelSelectors:
      "app": "kindnet"
  bandwidth:
    rate: 10mbps  
    limit: 50000  
    buffer: 50000
  duration: "19s"
  scheduler:
    cron: "@every 20s"
```
- inspect the test. It should complain that running this test is dangerous
- modify the chaos-controller and add the `-allowhostnetworktesting` switch:

```yaml
      ....
      containers:
      - command:
        - /usr/local/bin/chaos-controller-manager
        - -allowhostnetworktesting
      ...
```
- inspect the test again. Error message should be gone, test should proceed.
- I have also been able to confirm with tc on the node that this test is active:

```sh
root@kind-control-plane:/# tc qdisc  show dev eth0
qdisc tbf 1: root refcnt 2 rate 10485Kbit burst 49998b lat 0us  # ---> success
qdisc prio 2: parent 1: bands 3 priomap 1 2 2 2 1 2 0 0 1 1 1 1 1 1 1 1
qdisc sfq 4: parent 2:2 limit 127p quantum 1514b depth 127 divisor 1024 
qdisc sfq 5: parent 2:3 limit 127p quantum 1514b depth 127 divisor 1024 
qdisc sfq 3: parent 2:1 limit 127p quantum 1514b depth 127 divisor 1024 
root@kind-control-plane:/# 
```

Side effects


Related changes


### Does this PR introduce a user-facing change?


```release-note
This enables pods using host networking to be tested. Not for production use.
```
